File: examples/csharp/Scripts/MainMenu.cs
using Godot;

namespace FairyGUI.Examples;

public partial class MainMenu : DemoSceneBase
{
    protected override void ContinueInit()
    {
        UIPackage.AddPackage("res://Resources/UI/MainMenu");
        _view = UIPackage.CreateObject("MainMenu", "Main");
        _groot.AddChild(_view);

        NavBtn("n1", "res://csharp/Scenes/BasicsScene.tscn");
        NavBtn("n2", "res://csharp/Scenes/TransitionDemoScene.tscn");
        NavBtn("n4", "res://csharp/Scenes/VirtualListScene.tscn");
        NavBtn("n5", "res://csharp/Scenes/LoopListScene.tscn");
        NavBtn("n6", "res://csharp/Scenes/HitTestScene.tscn");
        NavBtn("n7", "res://csharp/Scenes/PullToRefreshScene.tscn");
        NavBtn("n8", "res://csharp/Scenes/ModalWaitingScene.tscn");
        NavBtn("n9", "res://csharp/Scenes/JoystickScene.tscn");
        NavBtn("n10", "res://csharp/Scenes/BagScene.tscn");
        NavBtn("n11", "res://csharp/Scenes/ChatScene.tscn");
        NavBtn("n12", "res://csharp/Scenes/ListEffectScene.tscn");
        NavBtn("n13", "res://csharp/Scenes/ScrollPaneScene.tscn");
        NavBtn("n14", "res://csharp/Scenes/TreeViewScene.tscn");
        NavBtn("n15", "res://csharp/Scenes/GuideScene.tscn");
        NavBtn("n16", "res://csharp/Scenes/CooldownScene.tscn");
    }

    private void NavBtn(string childName, string scenePath)
    {
        var btn = _view.GetChild(childName);
        if (btn == null)
            return;
        btn.AddClickListener(Callable.From(() =>
        {
            CleanupGrootOverlays();
            _groot.RemoveChildren();
            GetTree().ChangeSceneToFile(scenePath);
        }));
    }

    protected override void OnClose()
    {
        GetTree().Quit();
    }
}

File: examples/ts/Scripts/MainMenu.ts
/// <reference path="../fairygui.d.ts" />

import { Callable } from "godot";
import { DemoSceneBase } from "./DemoSceneBase";

export default class MainMenu extends DemoSceneBase {
    protected ContinueInit(): void {
        UIPackage.addPackage("res://Resources/UI/MainMenu");
        this._view = UIPackage.createObject("MainMenu", "Main");
        this._groot!.addChild(this._view);

        this._navBtn("n1", "res://ts/Scenes/BasicsScene.tscn");
        this._navBtn("n2", "res://ts/Scenes/TransitionDemoScene.tscn");
        this._navBtn("n4", "res://ts/Scenes/VirtualListScene.tscn");
        this._navBtn("n5", "res://ts/Scenes/LoopListScene.tscn");
        this._navBtn("n6", "res://ts/Scenes/HitTestScene.tscn");
        this._navBtn("n7", "res://ts/Scenes/PullToRefreshScene.tscn");
        this._navBtn("n8", "res://ts/Scenes/ModalWaitingScene.tscn");
        this._navBtn("n9", "res://ts/Scenes/JoystickScene.tscn");
        this._navBtn("n10", "res://ts/Scenes/BagScene.tscn");
        this._navBtn("n11", "res://ts/Scenes/ChatScene.tscn");
        this._navBtn("n12", "res://ts/Scenes/ListEffectScene.tscn");
        this._navBtn("n13", "res://ts/Scenes/ScrollPaneScene.tscn");
        this._navBtn("n14", "res://ts/Scenes/TreeViewScene.tscn");
        this._navBtn("n15", "res://ts/Scenes/GuideScene.tscn");
        this._navBtn("n16", "res://ts/Scenes/CooldownScene.tscn");
    }

    private _navBtn(childName: string, scenePath: string): void {
        const btn = this._view!.getChild(childName);
        if (btn != null) {
            btn.addClickListener(
                Callable.create(() => {
                    (this as unknown as { callDeferred(method: string, scenePath: string): void }).callDeferred(
                        "_navigateToScene",
                        scenePath,
                    );
                }),
            );
        }
    }

    _navigateToScene(scenePath: string): void {
        this._requestSceneChange(scenePath);
    }

    _onClose(): void {
        this.safeGetTree()?.quit();
    }
}
